Novo pull request

LindleyWerner opened the pull request.

dangon1 committed:
Merge remote-tracking branch 'refs/remotes/fegemo/master'

dangon1 committed:
Merge remote-tracking branch 'refs/remotes/origin/master' into fegemo/master

LindleyWerner committed:
Criação de mais um minijogo

dangon1 committed:
Implementacao novo jogo: NinjaTooth

dangon1 committed:
Ninja tooth exclusão de comentários e imagens desnecessárias

LindleyWerner committed:
Retirada de divergencias para o pull request

fegemo commented on core/src/br/cefetmg/games/minigames/SideWalking.java:
```diff
@@ -0,0 +1,174 @@
+    public void onUpdate(float dt) {
+        if (dente.getY() > alturachao) {
+            if (pulando) {
+                dente.setY(dente.getY() + velocidadePulo);
```
O pulo não ficou legal :cry:. Vale muito a pena implementar uma gravidadezinha!!
Com poucas linhas de código, vocês conseguem fazê-lo, se quiserem. A ideia é:
1. A classe `Teeth` vai ter um campo `float speedY`
2. Quando saltar (dentro do método `onHandlePlayingInput`), defina `dente.speedY = VELOCIDADE_INICIAL_SALTO` (um valor de 450f ficou legal aqui)
3. Na função `onUpdate`, se estiver `if (pulando)`:
   1. Ache a nova posição Y do `dente` baseado na velocidade Y (`dente.speedY`) e o `dt` (integração de Euler, como vimos na aula de hoje)
   2. Ache a nova `dente.speedY -= GRAVIDADE * dt` (um valor de 420f ficou legal aqui).
   3. Em vez de verificar se o dente bateu em cima para colocar `pulando = false`, verifique se o dente já pulou e voltou abaixo da altura do chão (`alturachao`).

Veja um trecho do código que adaptei:
![image](https://cloud.githubusercontent.com/assets/1650402/19992337/21fed3d0-a224-11e6-9545-f79ff7cae6f0.png)

fegemo commented on core/src/br/cefetmg/games/minigames/factories/SideWalkingFactory.java:
```diff
@@ -0,0 +1,41 @@
+public class SideWalkingFactory implements MiniGameFactory{
+     @Override
+    public MiniGame createMiniGame(BaseScreen screen, GameStateObserver observer, float difficulty) {
+        //throw new UnsupportedOperationException("Not supported yet."); //To change body of generated methods, choose Tools | Templates.
```
Limpar isto aqui

fegemo commented on core/src/br/cefetmg/games/minigames/SideWalking.java:
```diff
@@ -0,0 +1,174 @@
+            Sprite mst = enemies.get(i);
+            mst.setX(mst.getX() - velocidade);
+            if (dente.getBoundingRectangle().overlaps(mst.getBoundingRectangle())) {
+                backSound.stop();
```
Quando o jogo termina com vitória, a música de fundo continua tocando.
Analisei e não há atualmente uma forma legal disponível pra resolver isso sem alterar a classe `MiniGame`.

Sendo assim, peço que façam a seguinte alteração em `MiniGame.java`:
- Criem um método protegido com nome `onMiniGameFinished`, recebendo um parâmetro do tipo MiniGameState`. Esse método deve possuir uma implementação com um corpo vazio (abre e fecha chaves).
- Dentro de `transitionTo`, para os casos de `WIN` e `FAILED`, invoque esse novo método `onMiniGameFinished` passando `WIN` ou `FAILED`.
- No seu MiniGame, sobrescreva o método `onMiniGameFinished` e, dentro dele, encerre a música de fundo.

fegemo commented on core/src/br/cefetmg/games/minigames/factories/SideWalkingFactory.java:
```diff
@@ -0,0 +1,41 @@
+
+    @Override
+    public Map<String, Class> getAssetsToPreload() {
+        //throw new UnsupportedOperationException("Not supported yet."); //To change body of generated methods, choose Tools | Templates.
```
Limpar aqui também.

fegemo commented on core/src/br/cefetmg/games/screens/PlayingGamesScreen.java:
```diff
@@ -2,6 +2,25 @@
+import br.cefetmg.games.minigames.factories.PutTheBracesFactory;
+import br.cefetmg.games.minigames.factories.FleeTheTartarusFactory;
+import br.cefetmg.games.minigames.factories.CollectItensFactory;
+import br.cefetmg.games.minigames.factories.DentalKombatFactory;
```
O Matheus teve a ideia muito legal de incluir a linha 24 (logo abaixo) que importa todas as factories de uma só vez. Portanto, não é necessário, nem conveniente, voltar com as importações individuais.

fegemo reviewed (changes requested):
@LindleyWerner e @lcasviana, preciso de algumas alterações no MiniGame! Por favor, vejam os detalhes no código.

lcasviana committed:
Novas alterações...

fegemo committed:
Merging with upstream/master

fegemo committed:
Tarefa #29 (NinjaTooth).

Feita por Daniel.

lcasviana committed:
Corrigido bug do som.

lcasviana committed:
Bug som
Merge branch 'master' of https://github.com/fegemo/cefet-games-brushmyteethplz

lcasviana committed:
......
Merge branch 'master' of https://github.com/lcasviana/cefet-games-brushmyteethplz

lcasviana committed:
123123

lcasviana committed:
asdasd
Merge branch 'master' of https://github.com/lcasviana/cefet-games-brushmyteethplz

fegemo commented on core/src/br/cefetmg/games/minigames/SideWalking.java:
```diff
@@ -136,12 +213,35 @@ public String getInstructions() {
     public boolean shouldHideMousePointer() {
         return false;
     }
+<<<<<<< HEAD
```
Enviou conflitos sem resolução!!!
